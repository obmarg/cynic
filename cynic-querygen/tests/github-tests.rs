use insta::assert_snapshot;

use cynic_querygen::{document_to_fragment_structs, QueryGenOptions};

macro_rules! test_query {
    ($name:ident, $filename:literal) => {
        #[test]
        fn $name() {
            let schema = include_str!("../../schemas/github.graphql");
            let query = include_str!(concat!("queries/github/", $filename));

            assert_snapshot!(document_to_fragment_structs(
                query,
                schema,
                &QueryGenOptions::default()
            )
            .expect("QueryGen Failed"))
        }
    };
}

test_query!(add_comment_mutation, "add-comment-mutation.graphql");
test_query!(field_on_interface, "field-on-interface.graphql");
test_query!(fragment_on_interface, "fragment-on-interface.graphql");
test_query!(inline_fragment_on_union, "inline-fragment-on-union.graphql");
test_query!(
    inline_fragment_with_arguments,
    "inline-fragment-with-arguments.graphql"
);
test_query!(
    inline_fragment_with_renames,
    "inline-fragment-with-renames.graphql"
);
test_query!(input_object_arguments, "input-object-arguments.graphql");
test_query!(input_object_literals, "input-object-literals.graphql");
test_query!(literal_enums, "literal-enums.graphql");
test_query!(queries_with_typename, "queries-with-typename.graphql");

test_query!(issue_786, "issue-786.graphql");
