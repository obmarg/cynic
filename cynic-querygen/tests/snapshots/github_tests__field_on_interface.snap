---
source: cynic-querygen/tests/github-tests.rs
assertion_line: 31
expression: "document_to_fragment_structs(query, schema,\n        &QueryGenOptions::default()).expect(\"QueryGen Failed\")"
---
#[cynic::schema_for_derives(
    file = r#"schema.graphql"#,
    module = "schema",
)]
mod queries {
    use super::schema;

    #[derive(cynic::QueryFragment, Debug)]
    #[cynic(graphql_type = "Query")]
    pub struct FieldOnInterface {
        #[arguments(owner: "obmarg", name: "cynic")]
        pub repository: Option<Repository>,
    }

    #[derive(cynic::QueryFragment, Debug)]
    pub struct Repository {
        #[arguments(first: 1)]
        pub issues: IssueConnection,
    }

    #[derive(cynic::QueryFragment, Debug)]
    pub struct IssueConnection {
        pub edges: Option<Vec<Option<IssueEdge>>>,
    }

    #[derive(cynic::QueryFragment, Debug)]
    pub struct IssueEdge {
        pub node: Option<Issue>,
    }

    #[derive(cynic::QueryFragment, Debug)]
    pub struct Issue {
        pub author: Option<Actor>,
    }

    #[derive(cynic::QueryFragment, Debug)]
    pub struct Actor {
        pub login: String,
    }

}

mod schema {
    cynic::use_schema!(r#"schema.graphql"#);
}


